lib/extractor/docker-archive/index.ts

```diff
@@ -34,3 +38,35 @@ export function getPlatformFromConfig(
     ? `${imageConfig.os}/${imageConfig.architecture}`
     : undefined;
 }
+
+export function getDetectedLayersInfoFromConfig(imageConfig) {
+  const runInstructions = getUserInstructionLayersFromConfig(imageConfig)
+    .filter((instruction) => !instruction.empty_layer)
+    .map((instruction) => instruction.created_by.replace("# buildkit", ""));
+
+  const dockerfilePackages = getPackagesFromRunInstructions(runInstructions);
+  const dockerfileLayers = getLayersFromPackages(dockerfilePackages);
+  return { dockerfilePackages, dockerfileLayers };
+}
+
+export function getUserInstructionLayersFromConfig(imageConfig) {
+  const diffInHours = (d1, d2) => Math.abs(d1 - d2) / 1000 / (60 * 60);
+  const maxDiffInHours = 5;
+
+  const history = imageConfig.history;
+  if (!history) {
+    throw new Error("Failed to extract history from image config");
```

Comment:
what will happen in this case? will the entire scan fail? should we log something and return [] instead?